src/infoset.ts

```diff
@@ -46,8 +46,8 @@ async function openInfosetFilePrompt() {
 
     switch (action) {
       case 'Open':
-        let xml = await vscode.workspace.openTextDocument(uri)
-        await vscode.window.showTextDocument(xml, {
+        let data = await vscode.workspace.openTextDocument(uri)
```

Comment:
Do you really want to call this value "data" ? How about "infoset". Normally we think of data as the input to parse, and infoset as the output from parse, and most of your code follows that naming.